dialog/file.go

```diff
@@ -181,10 +184,36 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 
 	favorites := f.loadFavorites()
 
-	favoritesGroup := widget.NewGroup("Favorites", favorites...)
-	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, favoritesGroup, nil),
-		favoritesGroup, header, footer, body)
+	favoritesGroup := widget.NewVScrollContainer(widget.NewGroup("Favorites", favorites...))
+	var optionsButton *widget.Button
+	optionsButton = widget.NewButton("Options", func() {
+		f.optionsMenu(fyne.CurrentApp().Driver().AbsolutePositionForObject(optionsButton), optionsButton.Size())
+	})
+
+	left := fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, optionsButton, nil, nil), favoritesGroup, optionsButton)
```

Comment:
For future reference, `fyne.NewContainerWithLayout()` has been deprecated in favor of `container.New()`. However, you can use friendly container.NewBorder() constructor instead.

Reply:
Updated to container.NewBorder()